client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
+		const noticeId = `comment-notice-${ commentId }`;
+
+		dispatch( removeNotice( noticeId ) );
+
+		dispatch( successNotice( translate( 'Comment approved.' ), {
+			duration: 5000,
+			id: noticeId,
+			isPersistent: true,
+			button: translate( 'Undo' ),
+			onClick: () => dispatch( changeCommentStatus( siteId, postId, commentId, 'unapproved' ) )
```

Comment:
> About the whole "like+approve": since a "liked unapproved" comment doesn't really make sense, I've always worked with the impression that "undo like+approve" = "unlike+unapprove".

I find the whole split of the actions as confusing too. I really struggle to see the connection of undoing an action I didn't knowingly trigger. It's as if I made a change to a post and hit undo that then that undo unpublished the post.